sdk/identity/identity/samples/AzureIdentityExamples.md

```diff
@@ -0,0 +1,357 @@
+  const client = new SecretClient("https://key-vault-name.vault.azure.net", credential);
+}
+```
+
+## Authenticating a service principal with a client certificate
+
+This example demonstrates authenticating the `SecretClient` from the [@azure/keyvault-secrets][secrets_client_library] client library using the `ClientCertificateCredential`.
+
+For more information about how to create a service principal and get these values please refer to [Creating a Service Principal with the Azure CLI][service_principal_azure_cli].
+
+```ts
+/**
+ *  Authenticate with a client certificate.
+ */
+function withClientCertificateCredential() {
+  let credential = new ClientCertificateCredential(
+    "<YOUR_TENANT_ID>",
+    "<YOUR_CLIENT_ID>",
+    "<PATH_TO_CERTIFICATE>"
+  );
+  const client = new SecretClient("https://key-vault-name.vault.azure.net", credential);
+}
+```
+
+## Authenticating a user account with device code flow
+
+This example demonstrates authenticating the `SecretClient` from the [@azure/keyvault-secrets][secrets_client_library] client library using the `DeviceCodeCredential`.
+
+For more information about how to configure an AAD application for device code flow please refer to [Enable applications for device code flow][device_code_flow].
+
+```ts
+/**
+ *  Authenticate with a device code.
+ */
+function withDeviceCodeCredential() {
+  let credential = new DeviceCodeCredential(
+    process.env.AZURE_TENANT_ID!,
+    process.env.AZURE_CLIENT_ID!,
+    // In this scenario you may also omit this parameter since the default behavior is to log the message to the console
+    (deviceCodeInfo) => {
+      console.log(deviceCodeInfo.message);
+    }
+  );
+  const client = new SecretClient("https://key-vault-name.vault.azure.net", credential);
+}
+```
+
+## Authenticating a user account with username and password
+
+This example demonstrates authenticating the `SecretClient` from the [@azure/keyvault-secrets][secrets_client_library] client library using the `UsernamePasswordCredential`. The user must **not** have Multi-factor auth turned on.
+
+```ts
+/**
+ *  Authenticate with a client certificate.
+ */
+function withClientCertificateCredential() {
+  let credential = new UsernamePasswordCredential(
+    "<YOUR_TENANT_ID>",
+    "<YOUR_CLIENT_ID>",
+    "<USERNAME>",
+    "<PASSWORD>"
+  );
+  const client = new SecretClient("https://key-vault-name.vault.azure.net", credential);
+}
+```
+
+## Authenticating a user account interactively in the browser
+
+For clients that have a default browser available, the `InteractiveBrowserCredential` provides the most simple user authentication experience. In the sample below an application authenticates a `SecretClient` from the [@azure/keyvault-secrets][secrets_client_library] using the `InteractiveBrowserCredential`.
+
+```ts
+function withInteractiveBrowserCredential() {
+  const credential = new InteractiveBrowserCredential({
+    tenantId: "<YOUR_TENANT_ID>",
+    clientId: "<YOUR_CLIENT_ID>"
```

Comment:
What about commenting the unrequired properties? Not sure if that would be better. Up to you.